dadguide/monster_graph.py

```diff
@@ -102,9 +112,24 @@ def get_alt_cards(self, monster_id):
             ids.add(mid)
         return ids
 
-    def get_prev_evolution_by_monster(self, monster_id):
+    def get_prev_evolution_by_monster_id(self, monster_id):
         bes = self._get_edges(self.graph[monster_id], 'back_evolution')
         if bes: return bes.pop()
 
-    def get_next_evolutions_by_monster(self, monster_id):
+    def get_next_evolutions_by_monster_id(self, monster_id):
         return self._get_edges(self.graph[monster_id], 'evolution')
+
+    def is_monster_id_farmable(self, monster_id):
+        return self.graph.nodes[monster_id]['model'].is_farmable
+
+    def is_monster_farmable(self, monster: DgMonster):
+        return self.is_monster_id_farmable(monster.monster_no)
+
+    def is_monster_id_farmable_evo(self, monster_id):
+        for m in self.get_evo_tree(monster_id):
```

Comment:
you can make this a list comp: `any(m for m in self.get_evo_tree(monster_id) if self.is_monster_id_farmable(m))`

Reply:
done ty